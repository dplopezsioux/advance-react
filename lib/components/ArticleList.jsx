import React from "react";

import Article from "./Article";

const ArticleList = (props) => {
  return (
    <div>
      {console.log(props)}

      {Object.values(props.articles).map((article) => (
        <Article
          key={article.id}
          article={article}
          actions={props.articleAccions}
        />
      ))}
    </div>
  );
};

export default ArticleList;
