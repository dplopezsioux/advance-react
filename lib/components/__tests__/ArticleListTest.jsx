import React from "react";

import ArticleList from "../ArticleList";

import TestRenderer from "react-test-renderer";

describe("ArticleList", () => {
  const testProp = {
    article: {
      a: { id: "a" },
      b: { id: "b" },
    },
    articleAction: {
      lookupAuthor: jest.fn(() => ({})),
    },
  };

  it("Render Correctly", () => {
    const tree = TestRenderer.create(<ArticleList />).toJSON();
    console.log(tree);
  });
});
