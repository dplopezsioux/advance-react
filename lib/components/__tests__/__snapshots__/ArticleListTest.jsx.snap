// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList Render Correctly 1`] = `<div />`;
